src/test/java/org/littleshoot/proxy/TimeoutTest.java

```diff
@@ -132,9 +132,7 @@ public void testClientIdleBeforeRequestReceived() throws IOException, Interrupte
         // wait a bit to allow the proxy server to respond
         Thread.sleep(1500);
 
-        // the proxy should return an HTTP 504 due to the timeout
-        String response = SocketClientUtil.readStringFromSocket(socket);
-        assertThat("Expected to receive an HTTP 504 Gateway Timeout from the server", response, startsWith("HTTP/1.1 504"));
+        assertFalse("Client to proxy connection should be closed", SocketClientUtil.isSocketReadyToRead(socket));
```

Comment:
This is a change in behavior, connection is closed instead of sending timeout response. The idea is that proxy has not received a request yet and thus does not need to send any response.